src/js/strategies/DCVWebRTCStrategy.js

```diff
@@ -0,0 +1,148 @@
+        return this.isChromeBrowser() && this.getChromeBrowserVersion() >= CHROME_SUPPORTED_VERSION;
+    }
+
+    _createRtcPeerConnection(rtcPeerConnectionConfig, rtcPeerConnectionOptionalConfig) {
+        return this.proxy.createPeerConnection(rtcPeerConnectionConfig, rtcPeerConnectionOptionalConfig);
+    }
+
+    _buildMediaConstraints(self, mediaConstraints) {
+        if (self._enableAudio) {
+            var audioConstraints = {};
+            if (typeof self._echoCancellation !== UNDEFINED) {
+                audioConstraints.echoCancellation = !!self._echoCancellation;
+            }
+            if (Object.keys(audioConstraints).length > 0) {
+                mediaConstraints.audio = audioConstraints;
+            } else {
+                mediaConstraints.audio = true;
+            }
+        } else {
+            mediaConstraints.audio = false;
+        }
+    }
+
+    _gUM(constraints) {
+        return this.proxy.getUserMedia(constraints);
+    }
+
+    _createPeerConnection(configuration, optionalConfiguration) {
+        return this.proxy.createPeerConnection(configuration, optionalConfiguration);
+    }
+
+    addStream(_pc, stream) {
+        stream.getTracks().forEach(track => {
+            let transceiver = _pc.addTransceiver(track.kind, {
+                streams: [stream]
+            });
+            transceiver.sender.replaceTrack(track);
+        });
+    }
+
+    setRemoteDescription(self, rtcSession) {
+        var setRemoteDescriptionPromise = rtcSession._pc.setRemoteDescription(self._createSessionDescription({
+            type: ANSWER,
+            sdp: self._sdp
+        }));
+        setRemoteDescriptionPromise.catch(e => {
+            self.logger.error('SetRemoteDescription failed', e);
+        });
+        setRemoteDescriptionPromise.then(() => {
+            var remoteCandidatePromises = Promise.all(self._candidates.map(function (candidate) {
+                var remoteCandidate = self._createRemoteCandidate(candidate);
+                self.logger.info('Adding remote candidate', remoteCandidate);
+                return rtcSession._pc.addIceCandidate(remoteCandidate);
+            }));
+            remoteCandidatePromises.catch(reason => {
+                self.logger.warn('Error adding remote candidate', reason);
+            });
+            return remoteCandidatePromises;
+        }).then(() => {
+            rtcSession._sessionReport.setRemoteDescriptionFailure = false;
+            self._remoteDescriptionSet = true;
+            self._checkAndTransit();
+        }).catch(() => {
+            rtcSession._stopSession();
+            rtcSession._sessionReport.setRemoteDescriptionFailure = true;
+            self.transit(new FailedState(rtcSession, RTC_ERRORS.SET_REMOTE_DESCRIPTION_FAILURE));
+        });
+    }
+
+    onIceStateChange(evt, _pc) {
+        return _pc.iceConnectionState;
+    }
+
+    onPeerConnectionStateChange(_pc) {
+        return _pc.connectionState;
+    }
+
+    connect(self) {
```

Comment:
sorry this method is no longer needed for Citrix VDI, post [this commit](https://github.com/aws/connect-rtc-js/commit/87661287bfa65415416f792e37398847e7312992) so that self._strategy.connect is no longer called in rtc_session.js